bundles/org.openhab.core.model.rule/src/org/openhab/core/model/rule/jvmmodel/RulesRefresher.java

```diff
@@ -29,51 +49,133 @@
  * @author Kai Kreuzer - added delayed execution
  * @author Maoliang Huang - refactor
  */
-public class RulesRefresher {
+@Component(immediate = true, service = {})
+public class RulesRefresher implements ReadyTracker {
 
-    // delay before rule resources are refreshed after items or services have changed
-    private static final long REFRESH_DELAY = 2000;
+    // delay in ms before rule resources are refreshed after items or services have changed
+    private static final long REFRESH_DELAY = 5000;
+
+    private static final String POOL_NAME = "automation";
+    public static final String RULES_REFRESH = "rules_refresh";
 
     private final Logger logger = LoggerFactory.getLogger(RulesRefresher.class);
 
-    ModelRepository modelRepository;
-    private ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
     private ScheduledFuture<?> job;
+    private ScheduledExecutorService scheduler = ThreadPoolManager.getScheduledPool(POOL_NAME);
+    private boolean started;
+    private ReadyMarker marker = new ReadyMarker("dsl", RULES_REFRESH);
 
-    public void setModelRepository(ModelRepository modelRepository) {
-        this.modelRepository = modelRepository;
-    }
+    private ItemRegistryChangeListener itemRegistryChangeListener;
+    private ThingRegistryChangeListener thingRegistryChangeListener;
+
+    @Reference
+    protected ModelRepository modelRepository;
+
+    @Reference
+    protected ItemRegistry itemRegistry;
+
+    @Reference
+    protected ThingRegistry thingRegistry;
 
-    public void unsetModelRepository(ModelRepository modelRepository) {
-        this.modelRepository = null;
+    @Reference
+    protected EventPublisher eventPublisher;
+
+    @Reference
+    protected ReadyService readyService;
```

Comment:
Personally I never use `@Reference` on fields because it complicates writing unit tests or perhaps you know an easy way to inject these fields in tests?